org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JsonRpcHelpers.java

```diff
@@ -73,6 +112,41 @@ public static int toOffset(IDocument document, int line, int column) {
 		return toLine(toDocument(buffer), offset);
 	}
 
+	/**
+	 * Converts an offset to line number and column in an openable. If the
+	 * {@code openable} argument is {@link IOpenable#isOpen() <b>not</b> open}, this
+	 * method tries to
+	 * {@link IOpenable#open(org.eclipse.core.runtime.IProgressMonitor) open} it.
+	 *
+	 * @param buffer
+	 * @param line
+	 * @param column
+	 * @return
+	 */
+	public static int[] toLine(IOpenable openable, int offset) {
+		Assert.isNotNull(openable, "openable");
+		boolean mustClose = false;
+		try {
+			if (!openable.isOpen()) {
+				openable.open(new NullProgressMonitor());
+				mustClose = openable.isOpen();
+			}
+			IBuffer buffer = openable.getBuffer();
```

Comment:
Similar code to toOffset(), code could be refactored and generified, and a lambda be sent as a param (toLine or toOffset)